src/shimmerHttp.js

```diff
@@ -0,0 +1,208 @@
+import http from 'http';
+import https from 'https';
+import url from 'url';
+import { debuglog } from 'util';
+import shimmer from 'shimmer';
+import Perf from 'performance-node';
+import uuid from 'uuid/v4';
+import pickBy from 'lodash.pickby';
+import isArray from 'isarray';
+import { flatten } from 'flat';
+
+const debug = debuglog('@iopipe/trace');
+
+/*eslint-disable babel/no-invalid-this*/
+
+function unwrap() {
+  [http, https].forEach(mod => {
+    ['get', 'request'].forEach(method => {
+      if (mod[method].__wrapped) {
+        shimmer.unwrap(mod, method);
+      }
+    });
+    delete mod.__iopipeShimmer;
+  });
+}
+
+function wrapHttpGet(mod) {
+  return (options, cb) => {
+    const req = mod.request(options, cb);
+    req.end();
+    return req;
+  };
+}
+
+// these are keys that are mostly node specific and come from the actual js request object
+const unnecessaryReqKeys = [
+  'agent',
+  'automaticFailover',
+  'cache',
+  'decompress',
+  'followRedirect',
+  'host',
+  'href',
+  'retries',
+  'slashes',
+  'search',
+  'strictTtl',
+  'throwHttpErrors',
+  'useElectronNet',
+  'user-agent'
+];
+
+function excludeUnnecessaryReqKeys(obj) {
+  return pickBy(obj, (v, k) => unnecessaryReqKeys.indexOf(k) === -1);
+}
+
+function getReqDataObject(rawOptions) {
+  // options might be a string ie http.get(), coerce to object
+  const data =
+    typeof rawOptions === 'string'
+      ? { href: rawOptions, headers: {} }
+      : Object.assign({}, rawOptions, { href: url.format(rawOptions) }) || {};
+
+  // ensure url key is present with full URI
+  data.url = data.href;
+  // simple rename
+  data.query = data.search;
+
+  // sometimes request headers come in as an array
+  // make them strings to conform to our schema better
+  Object.keys(data.headers || {}).forEach(k => {
+    data.headers[k] = isArray(data.headers[k])
+      ? data.headers[k].join(' ')
```

Comment:
We might want to delimiter this in some way to make it clear it's not just one value.

Reply:
@kolanos I think that's fair, according to https://nodejs.org/api/http.html#http_request_setheader_name_value
you use an array to specify multiple headers that have the same key.
So it may be useful to us to make a clearly iopipe-specific delimiter in order to properly display the "duplicate" headers on the dashboard for example. Do you think `IOPIPE_DELIM` works? I'm not sure I want to use a standard delim like a comma since a user might already be using something like that?

Reply:
I like the `response.headers.Set-Cookie.0` paradigm. Let's go with that.